app/src/main/java/ch/epfl/sdp/item/ItemsViewAdapter.java

```diff
@@ -43,11 +43,15 @@ public void onBindViewHolder(@NonNull ItemsViewHolder holder, int position) {
         holder.name.setText(itemName);
         holder.amountOfItem.setText(String.valueOf(items.get(itemName)));
         holder.button.setOnClickListener(v -> {
+
+            itemFactory.getItem(itemName).useOn(PlayerManager.getInstance().getCurrentUser());
```

Comment:
Why move it to here? My question is for example, to use the health pack,  before the health is added on server-side and push to firebase, the Client only listens to the change of health point. But if you move it here, does it mean the health point will be added on Client-side silently and the server will not be aware of it. which will cause the problem when the server calculates the health point for the client based on the attack of the enemy?

Reply:
Yes, if the player's health gets to 0 on the server and the player uses a healthpack at the same time, the player will think it survived while it didn't. However I think the update will be fast enough for the player to not be aware of that.

Reply:
So now both server and client will update the health point in the cloud firebase? Will it be the problem?

Reply:
No only the server will.